OpenRA.Game/CPos.cs

```diff
@@ -42,6 +42,7 @@ public CPos(int x, int y, byte layer)
 		}
 
 		public static readonly CPos Zero = new CPos(0, 0, 0);
+		public static readonly CPos Invalid = new CPos(-2048, -2048, 255);
```

Comment:
Using magic values on CPos has caused many headaches in the past. Could we instead change `CPos IPathSearch.Expand` to `bool IPathSearch.TryExpand(out CPos)` and avoid this completely?